### DISCUSSION TITLE
App TODO formação Flutter - Compatibilidade

### DISCUSSION THREAD

**Discussion by leandesenv** [balta-io/forum#460] [Tira Dúvidas] [answered]
Boa tarde mestre @andrebaltieri !

Estou com um problema para rodar o fonte da aula Todo App do módulo Criando seu primeiro app com flutter. Erro:

PS C:\projetos\estudos\flutter\7184> dart pub upgrade
Resolving dependencies... 
The lower bound of "sdk: '>=2.1.0 <=3.0.0'" must be 2.12.0'
or higher to enable null safety.

The current Dart SDK (3.2.3) only supports null safety.

For details, see https://dart.dev/null-safety

Sou bem novato em flutter, mas saquei que o problema é de versões incompatível. Tenho instalado no meu note a ultima versão do flutter, sdk android, dart, porque peguei um projeto aqui no trabalho, mas to vendo suas aulas para aprender melhor.

Gostaria de saber se existe um gerenciador de versões tal como existe pro NPM onde a gente usa o nvm para chavear a versão do nodejs que queremos executar o projeto. Ah, existe alguma forma de realizar o upgrade do pubsec.yaml? Achei que o comando dart pub upgrade poderia resolver. 

Fiz o que dizia esse tutorial, mas não adiantou muito : https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects

Acredito que haja uma maneira melhor para resolver.

Obrigado balta!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @leandesenv, como vai?
Sugiro que utilize a mesma versão do SDK do Flutter que está no curso, tiveram mudanças nas últimas versões....

**Reply by leandesenv** [reply]
@andrebaltieri , vale a pena fazer esse curso? Quando iniciei pelos fundamentos de Dart, achei ok o Gabul apresentar o tema em 2022. Mas ao iniciar o módulo Criando sem primeiro App com flutter, percebi que está muito defasado. Vale a pena continuar? Quando tento rodar na minha máquina com o flutter mai atual, realmente existem muitos erros de sintaxe etc. Existe a possibilidade de refatorarem pelo menos o código fonte no github para que a gente tenha o projeto rodando no sdk mais atual, sejam eles dart/flutter, android. Iniciei uma migração aqui, mas a experiência não está ficando bacana. Para eu rodar no sdk sugerido no curso, como procedo? Extraio o flutter da versão do curso em alguma pasta e coloco na variável de ambiente PATH? Por fim, eu até me proponho a tentar realizar a migração e submeter um PR para sua análise.

Obrigado balta!

**Reply by andrebaltieri** [reply]
@leandesenv muitos conceitos são mantidos, mas as SDKs sofrem atualizações constantes, então sempre vão haver mudanças.
Porém na parte de layouts, setState e N outros itens se mantém... inclusive arquitetura, MVVM, etc...

Então a resposta é depende do que está procurando... se for algo mais conceitual sim, se for mais "how to", não!
